platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,246 @@
+- Remove forms library dependence on platform re-implementations of 3rd party functions, like lodash's `get` and `set`. Use the forms library implementations where possible.
+- Remove imports for `src/platform/utilities/environment` and instead rely on global `const`s from webpack's DefinePlugin for environment-related `const`s (i.e. an api url) or branching
+
+#### Package configuration
+
+**Estimate:** 3-5 days
+
+This is separate from updating the imports. We want the forms library not to depend on the "global" webpack & mocha configuration for `vets-website`, so we will add appropriate config files so that it can be built and tested as a standalone package.
+
+- CircleCI config to run forms library unit test on forms library changes
+- `vets-website` changes to config/scripts to prevent the forms library from getting build & tested as part of each PR
+
+
+#### Final touches (updating app code)
+
+By this point, the package will already exist, but due to some babel and webpack config it will be masquerading as if it were still spread across `platform/forms` and `platform/forms-system`. This is where we open a PR for each app team and update their code to import directly from the new package. Separate PRs for each app team will make each one easier to merge without requiring a codeowner review from every single team.
+
+- Update app code to use forms library JS & SCSS from the package instead of by direct path. This will help to establish the boundaries of the package for VFS teams.
+- Configure the `no-restricted-imports` ESLint rule to [restrict imports of `src/platform/packages` directly](https://eslint.org/docs/rules/no-restricted-imports)
+- Add an `.eslintrc` file to the new forms library package and configure `no-restricted-imports` to block imports from application code and platform code.
+- Remove the `.babelrc` forms-library alias & custom SASS importer
+
+Once app teams approve these PRs that we create, we will be able to remove those temporary configs and add an ESLint rule to prevent direct imports from `platform/packages/forms-library` in order to make sure that the forms library is treated like a package moving forward.
+
+Actual active time here will be fairly low (1-2 days), and will involve waiting for app teams before we will be able to finish the platform changes.
+
+### Alternatives
+
+See the section [Separate repo](#separate-repo) in Future Work. We are avoiding turning the forms library into its own node module with a separate repo because of the additional overhead of:
+
+- publishing the package
+- setting up CI pipelines
+- additional repo management.
+
+This _may_ be a path for the future.
+
+Another reason why we don't want to do this right now is that we would lost the advantage of the ESLint configuration that is part of `vets-website`.
+Once this configuration is published we will be able to import it into any repo, and at that point it may make sense to move the forms library out of `vets-website`.
+
```

Comment:
Based on our conversation today, we still need to decide decide yarn workspaces vs path dependency. Once we make that decision, the alternative should go here.